### DISCUSSION TITLE
How to disable linting for Iced buffer/stdout

### DISCUSSION THREAD

**Discussion by formsandlines** [liquidz/vim-iced#404] [Show and tell]
I just wanted to share a code snippet for nvim config that might help some people who would also like to disable the unhelpful, annoying linting in the Iced buffer/stdout.

It was originally suggested by @liquidz in [this issue comment](https://github.com/liquidz/vim-iced/issues/382#issuecomment-1014137699) for vim-lsp (which I don’t use and thus cannot verify, but surely it’ll work):

```
call iced#hook#add('connected', {
      \ 'type': 'function',
      \ 'exec': {_ -> lsp#disable_diagnostics_for_buffer(bufnr('iced_stdout'))},
      \ })
```

I have adapted it to [Ale](https://github.com/dense-analysis/ale) (using clj-kondo) like this:

```
call iced#hook#add('connected', {
      \ 'type': 'function',
      \ 'exec': {_ -> ale#toggle#DisableBuffer(bufnr('iced_stdout'))},
      \ })
```

And to [CoC](https://github.com/neoclide/coc.nvim/) (to use with [clojure-lsp](https://github.com/clojure-lsp/clojure-lsp), which integrates clj-kondo) like this:

```
call iced#hook#add('connected', {
      \ 'type': 'function',
      \ 'exec': {_ -> CocAction('diagnosticToggleBuffer', bufnr('iced_stdout'))},
      \ })
```

**Comment by liquidz** [comment]
@formsandlines Great!
Would it be useful to add this to the document? (e.g. [tips](https://liquidz.github.io/vim-iced/#tips))

**Reply by formsandlines** [reply]
For me it certainly would have been useful, I don’t know if it is much of an issue for other people though.

A problem could be that the function calls used in the snippets could differ with changes in the linting libraries, so docs would have to be updated if that happens. On the other hand, interface changes should not be so frequent so I guess it is fine if people just report when changes happen.
